src/api/java/com/ldtteam/structurize/api/util/ItemStackUtils.java

```diff
@@ -47,22 +51,47 @@ private ItemStackUtils()
      */
     public static List<ItemStack> getItemStacksOfTileEntity(final CompoundNBT compound, final World world, final BlockPos pos)
     {
-        final List<ItemStack> items = new ArrayList<>();
         final TileEntity tileEntity = TileEntity.readTileEntity(world.getBlockState(pos), compound);
-        if (tileEntity instanceof LockableTileEntity)
+        if (tileEntity == null)
         {
-            for (int i = 0; i < ((LockableTileEntity) tileEntity).getSizeInventory(); i++)
+            return Collections.emptyList();
+        }
+        tileEntity.setWorldAndPos(world, pos);
+
+        final List<ItemStack> items = new ArrayList<>();
+        for (final IItemHandler handler : getItemHandlersFromProvider(tileEntity))
+        {
+            for (int slot = 0; slot < handler.getSlots(); slot++)
             {
-                final ItemStack stack = ((LockableTileEntity) tileEntity).getStackInSlot(i);
+                final ItemStack stack = handler.getStackInSlot(slot);
                 if (!ItemStackUtils.isEmpty(stack))
                 {
                     items.add(stack);
                 }
             }
         }
+
         return items;
     }
 
+    /**
+     * Method to get all the IItemHandlers from a given Provider.
+     *
+     * @param provider The provider to get the IItemHandlers from.
+     * @return A list with all the unique IItemHandlers a provider has.
+     */
+    @NotNull
+    public static Set<IItemHandler> getItemHandlersFromProvider(@NotNull final ICapabilityProvider provider)
+    {
+        final Set<IItemHandler> handlerSet = new HashSet<>();
+        for (final Direction side : Direction.values())
+        {
+           provider.getCapability(ITEM_HANDLER_CAPABILITY, side).ifPresent(handlerSet::add);
```

Comment:
do chests/shulkers have this cap too? doesn't sideness cause duplication?

Reply:
It's a set to avoid duplication. If properly implemented it shouldn't cause it either.